### PR TITLE
main/cmake: upgrade to 3.8.0

### PR DESCRIPTION
**Pull request by pbregener**

### CODE REVIEW

File: main/cmake/APKBUILD [hunk 1]
```diff
@@ -41,6 +41,7 @@ build() {
 		--system-curl \
 		--system-expat \
 		--no-system-jsoncpp \
+		--no-system-librhash \
```
Comment (misery) [inline main/cmake/APKBUILD]: Maybe move from community to main? Otherwise it will build embedded rhash

https://git.alpinelinux.org/cgit/aports/tree/community/rhash

Comment (jirutka) [inline main/cmake/APKBUILD]: I’ve moved rhash to main. @pbregener, please update this PR to use system-provided rhash.

Comment (jirutka) [inline main/cmake/APKBUILD]: Also can we use system-provided jsoncpp?

Comment (misery) [inline main/cmake/APKBUILD]: jsoncpp needs cmake...
https://git.alpinelinux.org/cgit/aports/commit/main/cmake/APKBUILD?id=948b166b7599237fbf24af6577e0127ec842154f

Comment (jirutka) [inline main/cmake/APKBUILD]: Aha, understood.

Comment (jirutka) [inline main/cmake/APKBUILD]: Hm, rhash pkg needs some improvements.

Comment (pbregener) [inline main/cmake/APKBUILD]: Yep, will undo the change removing `--no-system-jsoncpp` and wait for `rhash` to be fixed.

Comment (misery) [inline main/cmake/APKBUILD]: Oops, sorry.... That is outdated. Jsoncpp uses scons now.

https://git.alpinelinux.org/cgit/aports/tree/main/jsoncpp/APKBUILD

**Comment by misery** [comment]
Added https://github.com/alpinelinux/aports/pull/1244 ... try rhash-dev if PR is merged. :-)

**Comment by pbregener** [comment]
Will do @misery. Maybe you can have a look at #1244 @jirutka?

**Comment by pbregener** [comment]
Alright, the `rhash-dev` package seems to work @misery! Now we just lack `jsoncpp-dev`..

**Comment by misery** [comment]
You should squash your commits. :-)

**Comment by pbregener** [comment]
Sure, will do once the build is green. I'm on a train right now and the internet connection is intermittent at best.

**Comment by pbregener** [comment]
Squashed. CI passed. Should be good to go now.

**Comment by jirutka** [comment]
Are you sure that it uses system-provided librhash now? Shouldn’t you add `--system-rhash`? I don’t see it in detected dependencies:

```
>>> cmake*: Tracing dependencies...
        so:libarchive.so.13
	so:libc.musl-x86_64.so.1
	so:libcurl.so.4
	so:libexpat.so.1
	so:libformw.so.6
	so:libgcc_s.so.1
	so:libjsoncpp.so.0
	so:libncursesw.so.6
	so:libstdc++.so.6
	so:libuv.so.1
	so:libz.so.1
```

Or it just links it statically?

**Comment by pbregener** [comment]
Looks like a static link:

Build log, line 617: `-- Using system-installed LIBRHASH`  
Build log, line 620: `-- Found LibRHash: /usr/lib/librhash.a`

**Comment by pbregener** [comment]
I guess one could shorten the `APKBUILD` by removing almost all `--system-XXX` lines, as `--system-libs` sets all other `--system-XXX` flags:
```
Usage: ./bootstrap [<options>...] [-- <cmake-options>...]
Options: [defaults in brackets after descriptions]
Configuration:
  --help                  print this message
  --version               only print version information
  --verbose               display more information
  --parallel=n            bootstrap cmake in parallel, where n is
                          number of nodes [1]
  --enable-ccache         Enable ccache when building cmake
  --init=FILE             load FILE as script to populate cache
  --system-libs           use all system-installed third-party libraries
                          (for use only by package maintainers)
  --no-system-libs        use all cmake-provided third-party libraries
                          (default)
  --system-curl           use system-installed curl library
  --no-system-curl        use cmake-provided curl library (default)
  --system-expat          use system-installed expat library
  --no-system-expat       use cmake-provided expat library (default)
  --system-jsoncpp        use system-installed jsoncpp library
  --no-system-jsoncpp     use cmake-provided jsoncpp library (default)
  --system-zlib           use system-installed zlib library
  --no-system-zlib        use cmake-provided zlib library (default)
  --system-bzip2          use system-installed bzip2 library
  --no-system-bzip2       use cmake-provided bzip2 library (default)
  --system-liblzma        use system-installed liblzma library
  --no-system-liblzma     use cmake-provided liblzma library (default)
  --system-libarchive     use system-installed libarchive library
  --no-system-libarchive  use cmake-provided libarchive library (default)
  --system-librhash       use system-installed librhash library
  --no-system-librhash    use cmake-provided librhash library (default)
```
